sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala

```diff
@@ -327,14 +327,24 @@ final class DataFrameWriter[T] private[sql](ds: Dataset[T]) {
                 supportsExtract, catalogManager, dsOptions)
 
               val location = Option(dsOptions.get("path")).map(TableCatalog.PROP_LOCATION -> _)
-
+              val tableSpec = TableSpec(
+                bucketSpec = None,
+                properties = Map(TableCatalog.PROP_PROVIDER -> source) ++ location,
```

Comment:
I think this should not include the location property. We already specify it in `TableSpec.location`

Reply:
Actually, we should put `Map.empty` here. provide is also specified in `TableSpec.provider`